File: tests/core/core_resource/main_test.cs
using System;
using System.IO;
using System.Reflection;
using System.Text;

using Xunit;

namespace example_test
{
    public class MyTest
    {
        [Fact]
        public void CheckData1()
        {
            var stream = typeof(MyTest).GetTypeInfo().Assembly.GetManifestResourceStream("example_test.data1.txt");
            var reader = new StreamReader(stream, Encoding.UTF8);
            var txt = reader.ReadToEnd();
            Assert.Equal("data1", txt);
        }
        [Fact]
        public void CheckDataMulti1()
        {
            var stream = typeof(MyTest).GetTypeInfo().Assembly.GetManifestResourceStream("example_test2.data1.txt");
            var reader = new StreamReader(stream, Encoding.UTF8);
            var txt = reader.ReadToEnd();
            Assert.Equal("data1", txt);
        }

        [Fact]
        public void CheckDataMulti2()
        {
            var stream = typeof(MyTest).GetTypeInfo().Assembly.GetManifestResourceStream("example_test2.data2.txt");
            var reader = new StreamReader(stream, Encoding.UTF8);
            var txt = reader.ReadToEnd();
            Assert.Equal("data2", txt);
        }
    }
}

File: tests/core/core_resource/main_test.fs
module example_test

open System
open System.IO
open System.Reflection
open System.Text

open Xunit


[<Fact>]
let ``CheckData1`` () =
    use stream =
        Assembly
            .GetExecutingAssembly()
            .GetManifestResourceStream("example_test.data1.txt")

    use reader = new StreamReader(stream, Encoding.UTF8)
    let txt = reader.ReadToEnd()
    Assert.Equal("data1", txt)


[<Fact>]
let ``CheckDataMulti1`` () =
    use stream =
        Assembly
            .GetExecutingAssembly()
            .GetManifestResourceStream("example_test2.data1.txt")

    use reader = new StreamReader(stream, Encoding.UTF8)
    let txt = reader.ReadToEnd()
    Assert.Equal("data1", txt)

[<Fact>]
let ``CheckDataMulti2`` () =
    use stream =
        Assembly
            .GetExecutingAssembly()
            .GetManifestResourceStream("example_test2.data2.txt")

    use reader = new StreamReader(stream, Encoding.UTF8)
    let txt = reader.ReadToEnd()
    Assert.Equal("data2", txt)